textarea/elfinder-sakai/src/java/org/sakaiproject/elfinder/sakai/SakaiFsService.java

```diff
@@ -194,7 +196,11 @@ public String getVolumeId(FsVolume volume) {
 			if("!admin".equals(currentSiteId) || "~admin".equals(currentSiteId)){
 				continue;
 			}
-			volumes.add(getSiteVolume(currentSiteId));
+			try
+			{
+				volumes.add(getSiteVolume(siteService.getSiteVisit(currentSiteId).getId()));
+			}
+			catch (PermissionException |IdUnusedException e) {}
```

Comment:
Instead of an empty catch block could you add a log.warn (or debug if to noisy) as it might come in handy for someone wondering why a volume was missing.